recipe/install-liblightgbm.sh

```diff
@@ -0,0 +1,50 @@
+# *****************************************************************
+#
+# Licensed Materials - Property of IBM
+#
+# (C) Copyright IBM Corp. 2020. All Rights Reserved.
+#
+# US Government Users Restricted Rights - Use, duplication or
+# disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+#
+# *****************************************************************
+#!/bin/bash
+
+export CMAKE_PREFIX_PATH=$PREFIX
+
+mkdir -p ${SRC_DIR}/build
+cd ${SRC_DIR}/build
+
+BUILD_OPTION=""
+# Determine Architecture
+ARCH="$(arch)"
+
+if [[ $build_type == "cuda" ]]
+then
+    BUILD_OPTION="-DUSE_CUDA=ON -DCMAKE_CUDA_COMPILER=${CUDA_HOME}/bin/nvcc -DCMAKE_CUDA_HOST_COMPILER=${CXX}"
+
+    # Create symlinks of cublas headers into CONDA_PREFIX
+    mkdir -p $CONDA_PREFIX/include
+    find /usr/include -name cublas*.h -exec ln -s "{}" "$CONDA_PREFIX/include/" ';'
+    export CXXFLAGS="${CXXFLAGS} -I${PREFIX}/include -I${CUDA_HOME}/include -I${CONDA_PREFIX}/include"
+
+    if [[ "${ARCH}" == 'ppc64le' ]]
+    then 
+        export CUDA_ARCH_FLAGS="3.7 6.0 7.0 7.5 8.0"
+    else
+        export CUDA_ARCH_FLAGS="3.7 5.2, 6.0 6.1 7.0 7.5 8.0"
```

Comment:
There are actually two difference ways to specify these:
 `sm_80`  vs `compute_80` where the latter includes PTX code which can be compiled forward JIT for newer hardware. It usually only makes sense to include that on the newest family.

I wonder how LightGBM uses these. Does it map each of these versions to `sm_` or `compute_` ?

Reply:
This is how LightGBM uses cuda compute flags - 
CUDA_ARCH_FLAGS: `arch=compute_60,code=sm_60 -gencode arch=compute_61,code=sm_61 -gencode arch=compute_62,code=sm_62 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75 -gencode arch=compute_75,code=compute_75`

Reply:
That seems a bit like overkill to me and will result in large binaries, but let's leave it for now. We can revisit when we centralize this stuff.

Reply:
I need to fix this cuda architecture flags properly. I checked the output and found out what we are setting isn't being propagated to the LightGBM build.

Reply:
LightGBM doesn't compile for 3.7 cuda compute capability.